windows/src/desktop/kmshell/main/Keyman.System.DownloadUpdate.pas

```diff
@@ -0,0 +1,278 @@
+(*
+  Name:             WebUpdateCheck
+  Copyright:        Copyright (C) SIL Global.
+  Documentation:
+  Description:
+  Create Date:      5 Dec 2023
+
+  Modified Date:
+  Authors:          rcruickshank
+  Related Files:
+  Dependencies:
+
+  Bugs:
+  Todo:
+  Notes:
+  History:
+*)
+
+unit Keyman.System.DownloadUpdate;
+
+interface
+uses
+  System.Classes,
+  System.SysUtils,
+  KeymanPaths,
+  httpuploader,
+  Keyman.System.UpdateCheckResponse,
+  OnlineUpdateCheck;
+
+const
+  CheckPeriod: Integer = 7; // Days between checking for updates
+
+type
+  TDownloadUpdateParams = record
+    TotalSize: Integer;
+    TotalDownloads: Integer;
+    StartPosition: Integer;
+  end;
+
+  TDownloadUpdate = class
+  private
+    FShowErrors: Boolean;
+    FDownload: TDownloadUpdateParams;
+    FErrorMessage: string;
+    {
+      Performs updates download in the background, without displaying a GUI
+      progress bar.
+      @params  SavePath  The path where the downloaded files will be saved.
+               Result    A Boolean value indicating the overall result of the
+               download process.
+    }
```

Comment:
I think as it is or `{*`  `}`. Looks like we don't say on the wiki

Reply:
I have seen the following style in places, which matches nicely with the javadoc patterns:

```delphi
(*
 * ....
 *)
```

But I think `///` matches Delphi's IDE patterns? Although not sure their doc integration has the same keywords and formatting...

Reply:
Did a bit of looking around the www. 
Delphi https://docwiki.embarcadero.com/RADStudio/Athens/en/Delphi_Comments
// for single line
{ } for modern style block
(* *) old style blocks
/// for XMLDoc where you can use xml tags
Then in terms on of Doxygen there is the usual array of started then no updated extensions for delphi. 
And each one did there own thing with the comments
/ =============================================================================
//! @file
//! @brief this is a Delphi source code template unit
//!
//! ...further Doxygen documentation for all entities in this Delphi unit...
//!
// =============================================================================
//! @brief DelphiDox template unit

Another did 
{*------------------------------------------------------------------------------
  Configure the User output and Debug output to inform about the status
  The TDOCProject class is very separated from the external user interface.
  But in order to inform about the state of the project, it needs to know where
  to display informations. This could be a visual control, or a file or whatever
  containing a TStrings reference to work.

  [@param](https://dephicodetodoc.sourceforge.net/JavaDoc_TAG_Param.htm) dmUser Where User messages will be written
  [@param](https://dephicodetodoc.sourceforge.net/JavaDoc_TAG_Param.htm) dmDebug Where Debug messages will be written
  [@see](https://dephicodetodoc.sourceforge.net/JavaDoc_TAG_See.htm)   WriteMessage
-------------------------------------------------------------------------------}